pkg/registry/apis/iam/user/store.go

```diff
@@ -64,23 +67,92 @@ func (s *LegacyStore) List(ctx context.Context, options *internalversion.ListOpt
 		return nil, err
 	}
 
+	if s.ac != nil {
+		return s.listWithCheck(ctx, ns, common.PaginationFromListOptions(options))
+	}
+
+	return s.listWithoutCheck(ctx, ns, common.PaginationFromListOptions(options))
+}
+
+func (s *LegacyStore) listWithCheck(ctx context.Context, ns claims.NamespaceInfo, p common.Pagination) (runtime.Object, error) {
+	ident, err := identity.GetRequester(ctx)
+	if err != nil {
+		return nil, err
+	}
+
+	check, err := s.ac.Compile(ctx, ident, claims.AccessRequest{
+		Verb:      "list",
+		Resource:  resource.GetName(),
+		Namespace: ns.Value,
+	})
+
+	if err != nil {
+		return nil, err
+	}
+
+	list := func(p common.Pagination) ([]iamv0.User, int64, int64, error) {
+		found, err := s.store.ListUsers(ctx, ns, legacy.ListUserQuery{
+			Pagination: p,
+		})
+
+		if err != nil {
+			return nil, 0, 0, err
+		}
+
+		out := make([]iamv0.User, 0, len(found.Users))
+		for _, u := range found.Users {
+			if check(ns.Value, strconv.FormatInt(u.ID, 10)) {
```

Comment:
👍🏻   would be really nice to soon replace:
https://github.com/grafana/grafana/blob/main/pkg/registry/apis/dashboard/legacy/sql_dashboards.go#L158. with the same pattern